controllers/controllers.go

```diff
@@ -33,6 +34,11 @@ const (
 	maxUnhealthyCount = 1
 )
 
+var (
+	// nodeLocker is used to synchronize updates across nodes from multiple controllers
+	nodeLocker sync.Mutex
```

Comment:
What was the result of looking into marking the controllers as busy and reusing that in this context also?

Reply:
Yes, I considered such scenarios, and I don't anticipate any effect on each other. 

The existing operator condition lock plays over the list of controllers. On the other hand,  the proposed lock synchronizes the API calls to get and set the upgrade annotation.